File: 268/MissingNumber.cs
/*
https://leetcode.com/problems/missing-number/description/?envType=daily-question&envId=2024-02-20
*/

public class Solution {

    public bool binarySearch(int[] nums, int val){
        int left=0, right=nums.Length-1;
        while(left<=right){
            int mid=left+(right-left)/2;
            if(nums[mid]==val) return true;
            else if(nums[mid]>val){
                right=mid-1;
            } else left=mid+1;
        }
        return false;
    }

    public bool find(int[] nums, int val){
        return binarySearch(nums, val);
    }

    public int MissingNumber(int[] nums) {
        Array.Sort(nums);
        int sz=nums.Length;
        for(int i=0; i<=sz; ++i){
            bool flag=find(nums, i);
            if(!flag) return i;
        }
        return -1;
    }
}

File: 2/268/MissingNumber.js
/*
https://leetcode.com/problems/missing-number/description/?envType=daily-question&envId=2024-02-20
*/

/**
 * @param {number[]} nums
 * @return {number}
 */

var binarySearch=function(nums, val){
    let left=0, right=nums.length-1;
    while(left<=right){
        let mid=Math.floor(left+(right-left)/2);
        if(nums[mid]===val) return true;
        else if(nums[mid]>val){
            right=mid-1;
        } else left=mid+1;
    }
    return false;
};

var find=function(nums, val){
    return binarySearch(nums, val);
};

var missingNumber = function(nums) {
    nums.sort((A, B)=>(A-B));
    let sz=nums.length;
    for(let i=0; i<=sz; ++i){
        let flag=find(nums, i);
        if(flag===false) return i;
    }
    return -1;
};